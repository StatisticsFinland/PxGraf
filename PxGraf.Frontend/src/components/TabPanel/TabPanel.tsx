import { Box } from '@mui/material';
import React from 'react';
import styled from 'styled-components';

interface ITabPanelProps {
    value: string | number;
    selectedValue: string | number;
    children: React.ReactNode;
}

const Wrapper = styled(Box)`
  padding: 24px;
`;

export const TabPanel: React.FC<ITabPanelProps> = (props) => {
    const { children, value, selectedValue, ...other } = props;

    return (
        <div
            role="tabpanel"
            hidden={selectedValue !== value}
            id={`simple-tabpanel-${value}`}
            aria-labelledby={`simple-tab-${value}`}
            {...other}
        >
            {selectedValue === value && (
                <Wrapper>
                    {children}
                </Wrapper>
            )}
        </div>
    );
}

export default TabPanel;