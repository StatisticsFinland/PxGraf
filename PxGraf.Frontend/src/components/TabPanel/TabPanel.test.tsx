import React from 'react';
import { render } from '@testing-library/react';
import { TabPanel } from './TabPanel';
import '@testing-library/jest-dom';

const mockChildren = ['foo', 'bar', 'baz'];

describe('Rendering test', () => {
    it('renders correctly with selected value', () => {
        // mock router for testing purposes
        const { asFragment } = render(<TabPanel value={'1'} selectedValue={'1'} children={mockChildren} />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('renders correctly with not selected value', () => {
        // mock router for testing purposes
        const { asFragment } = render(<TabPanel value={'1'} selectedValue={'2'} children={mockChildren} />);
        expect(asFragment()).toMatchSnapshot();
    });
});