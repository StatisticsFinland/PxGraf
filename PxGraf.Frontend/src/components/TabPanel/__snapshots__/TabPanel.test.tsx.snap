// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering test renders correctly with not selected value 1`] = `
<DocumentFragment>
  <div
    aria-labelledby="simple-tab-1"
    hidden=""
    id="simple-tabpanel-1"
    role="tabpanel"
  />
</DocumentFragment>
`;

exports[`Rendering test renders correctly with selected value 1`] = `
<DocumentFragment>
  <div
    aria-labelledby="simple-tab-1"
    id="simple-tabpanel-1"
    role="tabpanel"
  >
    <div
      class="sc-Qotzb dvSRyf MuiBox-root css-0"
    >
      foobarbaz
    </div>
  </div>
</DocumentFragment>
`;
